azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/DiskApiLiveTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.features;
+
+import java.net.URI;
+import java.util.List;
+
+import org.jclouds.azurecompute.arm.domain.CreationData;
+import org.jclouds.azurecompute.arm.domain.Disk;
+import org.jclouds.azurecompute.arm.domain.DiskProperties;
+import org.jclouds.azurecompute.arm.internal.BaseAzureComputeApiLiveTest;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.google.common.base.Predicate;
+
+import static org.jclouds.util.Predicates2.retry;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+@Test(groups = "live", singleThreaded = true)
+public class DiskApiLiveTest extends BaseAzureComputeApiLiveTest {
+
+   private String diskName;
+
+   @BeforeClass
+   @Override
+   public void setup() {
+      super.setup();
+      createTestResourceGroup();
+      diskName = "jclouds-" + RAND;
+   }
+
+   @Test
+   public void deleteDiskResourceDoesNotExist() {
+      assertNull(api().delete(diskName));
+   }
+
+   @Test(dependsOnMethods = "deleteDiskResourceDoesNotExist")
+   public void createDisk() {
+      DiskProperties properties = DiskProperties.builder().creationData(CreationData.create("Empty")).diskSizeGB(2).build();
+      Disk dataDisk = api().createOrUpdate(diskName, LOCATION, properties);
+      assertTrue(stateReached("Succeeded"), "start operation did not complete in the configured timeout");
+      assertTrue(dataDisk.properties().diskSizeGB() == 2);
+   }
+
+   @Test(dependsOnMethods = "createDisk")
+   public void getDisk() {
+      Disk dataDisk = api().get(diskName);
+      assertNotNull(dataDisk.name());
+      assertTrue(dataDisk.properties().diskSizeGB() == 2);
+   }
+
+   @Test(dependsOnMethods = "createDisk")
+   public void listDisks() {
+      List<Disk> dataDisks = api().list();
+      assertTrue(dataDisks.size() > 0);
```

Comment:
Verify that the created disk is there. We don't care about others? Just in case the RG is not empty.